src/HostCache.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Ably;
+
+// TODO - Add SyncMutex support to the class to avoid data corruption due to concurrent READ/WRITE (e.g. Apache multithreading environment)
```

Comment:
I have created an issue for this -> https://github.com/ably/ably-php/issues/154

Reply:
Nice, thanks, although I'm still a bit confused about the impact of this.. It sounds like if a user of the library is affected by this concurrent reading/writing behaviour they're gonna be worse off after upgrading to the new version of PHP? If that's the case then I think we should resolve #154 before releasing this.

Reply:
We have only refactored original implementation. So, it shouldn't really matter since no one reported this issue till now. Right now, the current implementation is robust enough to handle to concurrent caching. 
PS. I came with this scenario, because while working in golang we had to implement it.